### ISSUE TITLE
Link previews for images not clickable

### ISSUE THREAD

**Issue by paulchen** [RocketChat/Rocket.Chat#31861]
### Description:

When writing a message containing a link to an image file, a link preview will be shown. When hovering that preview, the mouse pointer changes to a hand symbol indicating that the preview is clickable. However, nothing happens when actually clicking the preview.

### Steps to reproduce:

1. Make sure link previews are enabled.
2. Submit a message containing a link to an image file.
3. Wait for the link preview to appear.
4. Click on the link preview.

### Expected behavior:

The image is shown in an overlay (as in Rocket.Chat 6.5.4).

### Actual behavior:

Nothing happens.

### Server Setup Information:

- Version of Rocket.Chat Server: 6.6.2
- Deployment Method: Docker
- Number of Running Instances: 1
- MongoDB Version: 6.0.13

### Client Setup Information

- Desktop App or Browser Version:
  - Electron client 3.9.14
  - Firefox 123.0 (64-bit)
  - Google Chrome Version 122.0.6261.95 (Offizieller Build) (64-Bit)
- Operating System: Windows 10

### Additional context

This is a regression. It works up to Rocket.Chat 6.5.4, but not in 6.6.0 and later.

EDIT: I'm unsure whether this is the same problem or a separate one, but images inserted using the [Tenor plugin](https://github.com/wreiske/Rocket.Chat.App-Tenor) are not clickable as well.

Maybe all of this is related to #31858.

**Comment by Himanshu-kiran** [comment]
Can you please provide image or video to make it more understandable

**Comment by paulchen** [comment]
> Can you please provide image or video to make it more understandable

No, I won't do that, sorry. In my opinion, the above description is sufficient to understand what the problem is.

By the way, the problem can also be reproduced on https://open.rocket.chat/.

**Comment by Himanshu-kiran** [comment]
> > Can you please provide image or video to make it more understandable
> 
> No, I won't do that, sorry. In my opinion, the above description is sufficient to understand what the problem is.
> 
> By the way, the problem can also be reproduced on https://open.rocket.chat/.
Code is written that way. Hovering on the link only preview the link the not any related image.Check this issue #31721

**Comment by XorZy** [comment]
I have the same problem on a brand new instance deployed with snap.
Thumbnails load fine but when you click on the image, it tries to load it from localhost:3000, which of course fails.
![image](https://github.com/RocketChat/Rocket.Chat/assets/57585415/49beef41-7761-4369-805e-3c972f0844ea)

**Comment by xcpdq** [comment]
We have the same issue with our Rocket Chat deployed via docker-compose (Enterprise license). Images from links are no longer clickable. Images sent as attachments are clickable.

**Comment by arjunrathod91** [comment]
Hey @paulchen,
Hope you are doing well. My name is Arjun and I am a mern stack developer. I thought that i can work on this issue. Can you please assign this to me? And please tell me steps that how i can setup this to my local.

**Comment by paulchen** [comment]
> I have the same problem on a brand new instance deployed with snap. Thumbnails load fine but when you click on the image, it tries to load it from localhost:3000, which of course fails.

@XorZy but your problem is about uploaded files, not linked images, right? In that case, there is a separate issue for that: #31858

> Can you please assign this to me? 

@arjunrathod91 I can't assign this issue to you, sorry. Firstly I'm not affiliated with Rocket.Chat. Secondly the Rocket.Chat team does not assign GitHub issues to external people. However, you are welcome to work on the issue and submit a pull request when you have fixed it.

> And please tell me steps that how i can setup this to my local.

I have no experience with Rocket.Chat development and have never set it up locally besides firing up a Docker image. However, there are [some instructions](https://github.com/RocketChat/Rocket.Chat#user-content-%EF%B8%8F-local-development) in README.md, maybe they suffice for your needs?

**Comment by SySagar** [comment]
I can confirm that all the issues related to this has been fixed in the latest version. You might want to update the rocket chat version.

**Comment by Himanshu-kiran** [comment]
Hii @SySagar,little off topic,
 I wanted to solve a issue related to "link preview on hovering" but I left
It will be helpful if you could  help me in finding required code/file(other than url preview) related to this.

**Comment by SySagar** [comment]
I am not exactly sure why we need a preview n hover until and unless it is a redirect. Also about this issue, like I said the current versions have it fixed.

**Comment by Himanshu-kiran** [comment]
Yes, it's not a issue so not guarantee it will be merged but can be good improvement i feel like I mean it's better to see the link preview near the link instead of end of messages.#31721

**Comment by paulchen** [comment]
> I can confirm that all the issues related to this has been fixed in the latest version. You might want to update the rocket chat version.

I'm sorry, it still doesn't work. I'm on 6.6.3 which is the most recent release of Rocket.Chat according to https://github.com/RocketChat/Rocket.Chat/releases and the issue persists.

**Comment by SySagar** [comment]
Oh sorry for the confusion it’s present in the latest development. Maybe we can wait till the latest production is released

**Comment by paulchen** [comment]
> Oh sorry for the confusion it’s present in the latest development. Maybe we can wait till the latest production is released

I just fired up the current `develop` image from Docker Hub on my local machine:

```
+----------------------------------------------+
|                SERVER RUNNING                |
+----------------------------------------------+
|                                              |
|  Rocket.Chat Version: 6.7.0-develop          |
|       NodeJS Version: 14.21.3 - x64          |
|      MongoDB Version: 6.0.14                 |
|       MongoDB Engine: wiredTiger             |
|             Platform: linux                  |
|         Process Port: 3000                   |
|             Site URL: http://localhost:3000  |
|     ReplicaSet OpLog: Enabled                |
|          Commit Hash: 6f9624b5fb             |
|        Commit Branch: develop                |
|                                              |
+----------------------------------------------+
```

This version is on par with the current `develop` branch: https://github.com/RocketChat/Rocket.Chat/commits/develop/

The issue persist in that version.

**Comment by eloo-abi** [comment]
Any news here? This bug is now open for nearly 1 year and it looks like nobody cares :( 
This bug is really problematic when you use RocketChat for alerts.

**Comment by eloo-abi** [comment]
i guess can be closed.. 
for link preview its working with the latest version

**Comment by paulchen** [comment]
> i guess can be closed.. for link preview its working with the latest version

I concur. I just verified that the problem described in this issue does not occur anymore as of Rocket.Chat 8.8.1. Therefore, I close this issue hereby.
